gobblin-runtime/src/main/java/org/apache/gobblin/runtime/messaging/hdfs/FileSystemMessageBuffer.java

```diff
@@ -16,47 +16,64 @@
  */
 package org.apache.gobblin.runtime.messaging.hdfs;
 
+import com.google.common.io.ByteStreams;
 import com.typesafe.config.Config;
+
 import java.io.IOException;
 import java.util.ArrayList;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Consumer;
 import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+
 import org.apache.commons.collections4.CollectionUtils;
 import org.apache.gobblin.runtime.messaging.MessageBuffer;
 import org.apache.gobblin.runtime.messaging.data.DynamicWorkUnitMessage;
 import org.apache.gobblin.runtime.messaging.data.DynamicWorkUnitSerde;
+import org.apache.gobblin.util.PathUtils;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FSDataOutputStream;
+import org.apache.hadoop.fs.FileStatus;
 import org.apache.hadoop.fs.FileSystem;
 import org.apache.hadoop.fs.Path;
 
+import static org.apache.gobblin.runtime.messaging.DynamicWorkUnitConfigKeys.*;
+
 
 /**
  * Implements {@link FileSystem} based message buffer for sending and receiving {@link DynamicWorkUnitMessage}.
  */
+@Slf4j
 public class FileSystemMessageBuffer implements MessageBuffer<DynamicWorkUnitMessage> {
+  private static final String messageFileSuffix = ".json";
+
   private boolean isPollingTaskScheduled = false;
 
   @Getter
   private final String channelName;
   private final FileSystem fs;
-  private final Path dir;
+  @Getter
```

Comment:
Minor nit: Change getter access level to Access level package since this getter exists only for the tests
- https://projectlombok.org/api/lombok/AccessLevel

Reply:
I think we can just leave this getter, as the path may be needed somewhere else. And it's a final variable so should be safe